network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/extensions/InjectionObservabilityImpl.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.network.store.iidm.impl.extensions;
+
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.iidm.network.Injection;
+import com.powsybl.iidm.network.extensions.InjectionObservability;
+import com.powsybl.iidm.network.extensions.ObservabilityQuality;
+import com.powsybl.network.store.iidm.impl.AbstractInjectionImpl;
+import com.powsybl.network.store.model.InjectionObservabilityAttributes;
+import com.powsybl.network.store.model.ObservabilityQualityAttributes;
+
+/**
+  * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public class InjectionObservabilityImpl<I extends Injection<I>> extends AbstractExtension<I> implements InjectionObservability<I> {
+
+    public InjectionObservabilityImpl(I injection) {
+        super(injection);
+    }
+
+    private AbstractInjectionImpl<?, ?> getInjection() {
+        return (AbstractInjectionImpl<?, ?>) getExtendable();
+    }
+
+    private InjectionObservabilityAttributes getInjectionObservabilityAttributes() {
+        return (InjectionObservabilityAttributes) getInjection().getResource().getAttributes().getExtensionAttributes().get(InjectionObservability.NAME);
+    }
+
+    public boolean isObservable() {
+        return getInjectionObservabilityAttributes().isObservable();
+    }
+
+    @Override
+    public InjectionObservability<I> setObservable(boolean observable) {
+        getInjection().updateResource(res -> ((InjectionObservabilityAttributes) res.getAttributes().getExtensionAttributes().get(InjectionObservability.NAME)).setObservable(observable));
```

Comment:
same, why don't you use the `getInjectionObservabilityAttributes` method instead of redoing this everywhere in this class

Reply:
Changed